### ISSUE TITLE
The error message when the file name and constructor don't match is not clear enough

### ISSUE THREAD

**Issue by javier-tarazaga** [SuperblocksHQ/superblocks-lab#10]
### Description
The error message when the contract file name and the constructor don't match is not clear enough, therefore making it really difficult to understand what the issue is. 

### Steps to reproduce
1. Create a new contract. 
2. Rename the contract in the code editor (and the constructor)
3. Compile the contract

### Expected result
The IDE would give you a more fine-grained error message telling you that the names do not match. 

### Actual result
The IDE simply returns "The contract could apparently not be compiled." which is not really useful to understand what the problem might be. 

### Reproducible
100%

**Comment by javier-tarazaga** [comment]
@filippsen is this something we could figure out when compiling the contract or actually the error message is produced by the compiler itself?

**Comment by filippsen** [comment]
@javier-tarazaga 

From my understanding, compiling a _Solidity_ file whose name doesn't match any contract declaration or definition is valid. It just happens that it is often referred to as good practice to have a single contract declaration and definition for each file, matching names (like _Java_, _C#_ and similar).

With that in mind, this issue will not produce compile errors (at the low level). Current code, however, does take the file name into consideration, when looking for the respective contract definition, and compilation process ends up failing with a generic error, as pointed out. The reason for that is due to the following line of code in `this.props.functions.compiler.queue` (`src/components/projecteditor/compiler.js`):

```
if(data.contracts) contractObj=data.contracts[filename][contractName];
```

Taking _Hello World_ as a base: if the contract definition is renamed to something else, for example, `contract HiWorld`, keeping the same file name (`HelloWorld.sol`), it results in the following:

* `data.contracts` will be valid
* `filename` will be `HelloWorld.sol`
* `contractName` will be set to HelloWorld

That causes `contractObj` to be set to `undefined`, because `contractName` key will be missing, leading into the "generic" error previously observed.

A solution for adding a more clear error message could be to include an extra check, with a fine-grained message when `contractObj` is `undefined`, in the `else` statement (when `contractObj` validity is checked):

```
if(Object.keys(data.contracts[filename]).length > 0) 
    // contractObj is in undesirable state
    // Compiled one or more contracts
    // Expected contractName to be present
    message = "[ERROR] Expected " + contractName + " to be present."
else
    message = "[ERROR] The contract " + contractName + " could not be compiled."
```

**Comment by futurejones** [comment]
This appears to have been fixed in commit [(bdbe8d7)](https://github.com/SuperblocksHQ/superblocks-lab/commit/bdbe8d75fa35d02445f35e73abdf89a8197a55be#diff-f9a759603de394a43278e8127bf15f78)

**Comment by filippsen** [comment]
About the constructor: using `function` keyword to define a constructor is deprecated since _Solidity_ version `0.4.22`. The `constructor` keyword should be used instead. According to the official changelog:
> Constructors should now be defined using constructor(uint arg1, uint arg2) { ... } to make them stand out and avoid bugs when contracts are renamed but not their constructors.  

Link: https://github.com/ethereum/solidity/releases/tag/v0.4.22

About the file name: renaming a contract file without updating its contents to reflect the contract name change will result in following compiler error:
```
[ERROR] The contract <name> could not be compiled. The contract needs to be named the same as the contract's source file.
```
Indeed, this has been fixed in version `1.0.1` (issue #104). Thank you @futurejones.
